besu/src/main/java/org/hyperledger/besu/RunnerBuilder.java

```diff
@@ -790,9 +790,18 @@ public Runner build() {
       LOG.debug("added ethash observer: {}", stratumServer.get());
     }
 
-    sanitizePeers(network, staticNodes)
-        .map(DefaultPeer::fromEnodeURL)
-        .forEach(peerNetwork::addMaintainedConnectionPeer);
+    if (besuController.getGenesisConfigOptions().isPoa()) {
+      // In a permissioned chain Besu should maintain connections to both static nodes and
+      // bootnodes, which includes retries periodically
+      Stream.concat(sanitizePeers(network, staticNodes), sanitizePeers(network, bootnodes))
```

Comment:
nit: could the stream be factored out as a variable and the map and forEach statements are only written once? This insulates us from changes to those steps accidentally only impacting one path.

Reply:
Sure, I'll make that change now and push a new commit

Reply:
See latest changes @shemnon